### PR TITLE
Updated German translation

### PR DESCRIPTION
**Pull request by phriedrich**
### PR Creator Checklist

Ensure you've checked the following before submitting your PR:

- [ ] You've discussed making your changes with a member of the dev team per contributing rules in the README
- [x] Your changes are free of any lint errors
- [x] Your changes are free of any typescript errors
- [x] You've tested your changes

### Summary

Updates and fixes for several strings in the German translation, also adding previously untranslated terms.

Apart from simple spelling mistakes, I have tried to fix grammatical errors. Where possible I have tried to use the same terms Lemmy's web interface is using to avoid confusion. 

For example, while the correct English plural for "community" is "communities", this doesn't apply when using the English word within a German text. The _recommendation_ ist to use the English singular form and add an _s_, resulting in "Communitys". Also note the capital first letter used in German for all nouns.

This may look a bit weird, but is the common form. Also the Lemmy team uses this in the translation for the web frontend, so we should try to keep it consistent.

### Screenshots

> If the UI has been changed, include screenshots. 
> We will not review your PR without screenshots if they are applicable.

### Test Plan

> Please documemnt the steps required to test your PR

### CODE REVIEW

**Comment by sgriff96** [comment]
@Felix-Staud can you take a look at this

**Comment by Felix-Staud** [comment]
Thanks for fixing all my translation mess ups. 
I had some small feedbacks. Please let me know when it's ready for re-review

**Comment by phriedrich** [comment]
@Felix-Staud Where can I see this feedback?

**Comment by Felix-Staud** [comment]
> @Felix-Staud Where can I see this feedback?

Inside the “files” tab you can see the exact lines I’ve commented

**Review by Felix-Staud — CHANGES_REQUESTED** [review]

File: src/plugins/i18n/locales/de.json [hunk 1]
```diff
@@ -15,43 +15,46 @@
...
     "unknownError": "Es ist ein unbekannter Fehler aufgetreten.",
     "userBlockedSuccess": "Benutzer erfolgreich blockiert.",
     "accentColorUpdated": "Akzentfarbe angepasst",
-    "accentColorInvalidHexCode": "Akzentfarbe ist kein valider Hex Farbcode",
+    "accentColorInvalidHexCode": "Akzentfarbe ist kein valider Hex-Farbcode",
     "allFieldsRequired": "Alle Felder müssen gefüllt sein.",
     "useUsername": "Bitte gib nur deinen Benutzernamen an (ohne Instanz/@).",
     "passwordsMismatch": "Die Passwörter stimmen nicht überein.",
-    "blocked": "Blocked {{0}}",
+    "blocked": "Blockiert {{0}}",
```
[the remark this answers is not in the document]
Comment (phriedrich) [reply src/plugins/i18n/locales/de.json:57]: Ah, I see. I'll make some fixes and will update.

File: src/plugins/i18n/locales/de.json [hunk 2]
```diff
@@ -94,67 +97,82 @@
...
+        "hide": "NSFW-Inhalt ausblenden"
       },
       "web": {
         "useReaderMode": "Lesemodus verwenden"
       },
       "markRead": {
-        "header": "Gelesen Markieren Einstellungen",
-        "onPostOpen": "Beim öffnen vom Post",
-        "onImageView": "Beim ansehen vom Bild",
+        "header": "Einstellungen für als gelesen Markieren",
```
[the remark this answers is not in the document]
Comment (phriedrich) [reply src/plugins/i18n/locales/de.json:108]: "Als gelesen Markieren" should do, when this is followed by the list, it should be clear enough.

File: src/plugins/i18n/locales/de.json [hunk 3]
```diff
@@ -340,25 +362,25 @@
...
   "Defederated": "Entföderiert",
-  "Registration Closed": "Registrierung Geschlossen",
+  "Registration Closed": "Registrierung geschlossen",
   "Application Required": "Bewerbung nötig",
-  "Open Registration": "Frei Zugänglich",
+  "Open Registration": "Frei zugänglich",
   "Subscribed": "Abonniert",
   "Subscribe": "Abonnieren",
-  "Post": "Pfostieren",
+  "Post": "Beitrag erstellen",
```
[the remark this answers is not in the document]
Comment (phriedrich) [reply src/plugins/i18n/locales/de.json:383]: Bildschirmfoto 2023-07-24 um 07 40 42

Together with the + in front this should be clear enough.

File: src/plugins/i18n/locales/de.json [hunk 4]
```diff
@@ -94,67 +97,82 @@
...
-        "useSystemFont": "System Schriftart übernehmen",
-        "useSystemFontSize": "System Schriftgröße übernehmen",
+        "useSystemFont": "System-Schriftart übernehmen",
+        "useSystemFontSize": "System-Schriftgröße übernehmen",
         "textSize": "Textgröße",
-        "postTitleFontWeight": "Schriftstärke - Post Titel"
+        "postTitleFontWeight": "Schriftstärke für Beitrags-Titel"
       },
 			"gestures": {
 				"header": "GESTURES",
```
[the remark this answers is not in the document]
Comment (phriedrich) [reply src/plugins/i18n/locales/de.json:147]: "Stärke für Beitrags-Titel"

File: src/plugins/i18n/locales/de.json [hunk 5]
```diff
@@ -340,25 +362,25 @@
...
+  "Captcha Answer": "Captcha-Antwort",
   "Link": "Link",
   "Login": "Einloggen",
   "Yes": "Ja",
   "No": "Nein",
   "Join": "Beitreten",
   "Hub Info": "Hub Info",
   "Other Info": "Weitere Infos",
-  "Legal Info": "Rechliche Infos",
+  "Legal Info": "Rechtliche Infos",
```
[the remark this answers is not in the document]
Comment (phriedrich) [reply src/plugins/i18n/locales/de.json]: Sounds good

File: src/plugins/i18n/locales/de.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Felix-Staud) [reply src/plugins/i18n/locales/de.json]: Agree

**Comment by phriedrich** [comment]
I think I got all

**Comment by Felix-Staud** [comment]
I think it looks good now, you have to resolve the merge conflicts though @phriedrich . _If you don’t know how to do that I can help you_

**Comment by phriedrich** [comment]
I think I got it

**Review by Felix-Staud — APPROVED** [review]
@sgriff96 LGTM

**Comment by Felix-Staud** [comment]
@phriedrich congrats on your first PR 🎉.

_danke für deine Unterstützung_

**Pull request merged**